test/cpp_test_suite/new_tests/cxx_cmd_types.cpp

```diff
@@ -879,24 +879,88 @@ class CmdTypesTestSuite: public CxxTest::TestSuite
     {
         for (size_t i = 0; i < loop; i++)
         {
-            DeviceData din, dout;
             DevicePipeBlob in{"TestPipeCmd"};
-            vector<string> names{"level0"};
+            DeviceData din, dout;
+            DeviceData din2, dout2;
+
+            vector <string> names{"long", "double", "string", "double array"};
             in.set_data_elt_names(names);
-            DevLong data = 123;
-            in << data;
-            din.insert(in);
-            TS_ASSERT_THROWS_NOTHING(dout = device->command_inout("IOPipeBlob", din));
-            DevicePipeBlob received{};
-            dout.extract(&received);
+            DevLong long_data = 123;
+            TS_ASSERT_THROWS_NOTHING(in["long"] << long_data);
+            DevDouble double_data = 3.14;
+            TS_ASSERT_THROWS_NOTHING(in["double"] << double_data);
+            string data_str = "String value";
+            TS_ASSERT_THROWS_NOTHING(in["string"] << data_str);
+            vector<double> double_array = {0.1, -0.02, 0.003, -0.0004};
+            TS_ASSERT_THROWS_NOTHING(in["double array"] << double_array);
+            TS_ASSERT_THROWS_NOTHING(din << in);
 
```

Comment:
If we do something like the following here, between the insert and the command_inout:
```
DevDouble double_data2 = 789.123;
in["double"] << double_data2;
```
Then, with the previous version of DeviceData::insert, it crashes.
With the [new version](https://github.com/tango-controls/cppTango/pull/378/commits/2ceeb7edeb841c7a06d823971b880cbbf3b0436c), the DeviceData will point to the modified DevicePipeBlob, meaning the data element named "double" will contain the new value, not the one which was in DevicePipeBlob object when the insert occurred.
We can discuss whether this is good or bad. If we stay with this solution which is similar to what is done for DevEncoded, we need to document this well and warn the users about this pitfall.
I think there will be a crash if the DevicePipeBlob object is deleted before the command_inout occurs. This should be very well documented too if we stay with this approach.
If we go for the more intuitive approach which would consist in copying everything from the DevicePipeBlob object when the insert into a DeviceData occurs, we will clearly degrade performances if the blob is big.